src/withAuthUserTokenAPI.js

```diff
@@ -0,0 +1,42 @@
+import { verifyIdToken } from 'src/firebaseAdmin'
+import logDebug from 'src/logDebug'
+
+/**
+ * API Middleware for Firebase ID Token Authorization.
+ * API endpoints wrapped with this middleware will be required to be executes with a
+ * `bearer <id token>` Authorization header.
+ * e.g.,
+ * ```
+ * await fetch('<baseUrl>/api/hello', { headers: {authorization: 'bearer <IDToken>'}});
+ * ```
+ * @param {*} handler
+ * @returns Wrapped handler method that is used by NextJS.
+ */
+const withAuthUserTokenAPI = (handler) => async (req, res) => {
+  if (
+    !req.headers.authorization ||
+    !req.headers.authorization.match(/^bearer .*$/i)
+  ) {
+    res.status(401).send({ message: 'Unauthorized', code: 1 })
+    return
+  }
+
+  const token = req.headers.authorization.split(' ')[1]
+
+  try {
+    const authUser = await verifyIdToken(token)
+
+    if (!authUser.id) {
```

Comment:
I think that I am not getting 100% how `verifyIdToken` works here. I assumed (when creating) that this method could possibly return a new user (with a null `id`); however I seem to not be able to get this to happen; @kmjennison are you able to cast your eye on these lines and provide some insight?

Reply:
Hi , I'm a newbie for this library , is this related to getting AuthUser in next js api route handler?

Reply:
So I've 'fixed' the test to hit the 2 lines that I was missing. 

However I am still unconvinced that the scenario that I am testing for is possible (with this usage of the `verifyToken` method). From what I can understand (which is limited) the `verifyIdToken` will throw and error if the ID Token is not valid and so there _should_ be no way that I get an `AuthUser` that does not have an `id` property set. 

Am I missing something?